src/test/javascript/spec/component/text-assessment-dashboard/text-assessment-dashboard.component.spec.ts

```diff
@@ -0,0 +1,257 @@
+    let fixture: ComponentFixture<TextAssessmentDashboardComponent>;
+    let exerciseService: ExerciseService;
+    let textSubmissionService: TextSubmissionService;
+    let textAssessmentService: TextAssessmentService;
+    let accountService: AccountService;
+    let sortService: SortService;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule],
+            declarations: [
+                TextAssessmentDashboardComponent,
+                MockComponent(AssessmentFiltersComponent),
+                MockComponent(AssessmentWarningComponent),
+                MockComponent(ResultComponent),
+                MockDirective(HasAnyAuthorityDirective),
+                MockDirective(TranslateDirective),
+                MockDirective(SortDirective),
+                MockRouterLinkDirective,
+                MockQueryParamsDirective,
+                TranslatePipeMock,
+                MockPipe(ArtemisDatePipe),
+            ],
+            providers: [
+                JhiLanguageHelper,
+                { provide: AccountService, useClass: MockAccountService },
+                { provide: ExerciseService, useClass: MockExerciseService },
+                MockProvider(TextSubmissionService),
+                MockProvider(TextAssessmentService),
+                MockProvider(SortService),
+                {
+                    provide: ActivatedRoute,
+                    useValue: {
+                        snapshot: {
+                            paramMap: convertToParamMap({
+                                exerciseId: textExercise.id,
+                                examId: 2,
+                            }),
+                        },
+                    },
+                },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(TextAssessmentDashboardComponent);
+                component = fixture.componentInstance;
+                exerciseService = TestBed.inject(ExerciseService);
+                textSubmissionService = TestBed.inject(TextSubmissionService);
+                textAssessmentService = TestBed.inject(TextAssessmentService);
+                accountService = TestBed.inject(AccountService);
+                sortService = TestBed.inject(SortService);
+            });
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    it('should set parameters and call functions on init', () => {
+        // setup
+        const exerciseServiceFindMock = jest.spyOn(exerciseService, 'find');
+        exerciseServiceFindMock.mockReturnValue(of(new HttpResponse({ body: textExercise })));
+        const getTextSubmissionStub = jest.spyOn(textSubmissionService, 'getTextSubmissionsForExerciseByCorrectionRound');
+        getTextSubmissionStub.mockReturnValue(of(new HttpResponse({ body: [textSubmission], headers: new HttpHeaders() })));
```

Comment:
This stuff can be inlined. Saves you a variable :)
Same goes for the spies you use below
```suggestion
        jest.spyOn(exerciseService, 'find').mockReturnValue(of(new HttpResponse({ body: textExercise })));
        const getTextSubmissionStub = jest.spyOn(textSubmissionService, 'getTextSubmissionsForExerciseByCorrectionRound').mockReturnValue(of(new HttpResponse({ body: [textSubmission], headers: new HttpHeaders() })));
```